godot-core/src/builtin/signal.rs

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::fmt;
+use std::ptr;
+
+use godot_ffi as sys;
+
+use super::meta::{impl_godot_as_self, FromGodot, GodotType};
+use crate::builtin::meta::ToGodot;
+use crate::builtin::{inner, Array, Callable, Dictionary, StringName, Variant};
+use crate::engine::{global::Error, Object};
+use crate::obj::bounds::DynMemory;
+use crate::obj::{Bounds, Gd, GodotClass, InstanceId};
+use sys::{ffi_methods, GodotFfi};
+
+/// A `Signal` represents a signal of an Object instance in Godot.
+///
+/// Signals are composed of a reference to an `Object` and the name of the signal on this object.
+#[repr(C, align(8))]
+pub struct Signal {
+    opaque: sys::types::OpaqueSignal,
+}
+
+impl Signal {
+    fn from_opaque(opaque: sys::types::OpaqueSignal) -> Self {
+        Self { opaque }
+    }
+
+    /// Create a signal for the signal `object::signal_name`.
+    ///
+    /// _Godot equivalent: `Signal(Object object, StringName signal)`_
+    pub fn from_object_signal<T, S>(object: &Gd<T>, signal_name: S) -> Self
+    where
+        T: GodotClass,
+        S: Into<StringName>,
+    {
+        let signal = signal_name.into();
+        unsafe {
+            sys::from_sys_init_or_init_default::<Self>(|self_ptr| {
+                let ctor = sys::builtin_fn!(signal_from_object_signal);
+                let raw = object.to_ffi();
+                let args = [raw.as_arg_ptr(), signal.sys_const()];
+                ctor(self_ptr, args.as_ptr());
+            })
+        }
+    }
+
+    /// Connects this signal to the specified callable.
+    ///
+    /// Optional flags can be also added to configure the connection's behavior (see [`ConnectFlags`](crate::engine::object::ConnectFlags) constants).
+    /// You can provide additional arguments to the connected callable by using `Callable::bind`.
+    ///
+    /// A signal can only be connected once to the same [`Callable`]. If the signal is already connected,
+    /// returns [`Error::ERR_INVALID_PARAMETER`] and
+    /// pushes an error message, unless the signal is connected with [`ConnectFlags::CONNECT_REFERENCE_COUNTED`](crate::engine::object::ConnectFlags::CONNECT_REFERENCE_COUNTED).
```

Comment:
```suggestion
    /// pushes an error message, unless the signal is connected with [`ConnectFlags::CONNECT_REFERENCE_COUNTED`](crate::engine::object::ConnectFlags::REFERENCE_COUNTED).
```

The other one is deprecated.